app/Jobs/DiscordAuditJob.php

```diff
@@ -164,4 +160,88 @@ public function handle(
             }
         }
     }
+
+    /**
+     * Performs audit by interating through HMS users, adding discord roles if they are missing.
+     *
+     * @param Discord $discord The discord helper object
+     * @param ProfileRepository $profileRepository Profile repository object
+     * @param RoleRepository $roleRepository Role repository object
+     *
+     * @return void
+     */
+    private function reverse(Discord $discord, ProfileRepository $profileRepository, RoleRepository $roleRepository)
+    {
+        $currentMembers = $roleRepository->findOneByName(Role::MEMBER_CURRENT)->getUsers();
+        $discordMembers = $discord->listGuildMembers();
+
+        // Build an array of roles in Discord which match roles in HMS.
+        $teams = $roleRepository->findAllTeams();
+        $auditableRoles = [
+            $discord->findRoleByName('Current Member'),
+        ];
+
+        foreach ($teams as $team) {
+            $discordRole = $discord->findRoleByName($team->getDisplayName());
+            if ($discordRole) {
+                array_push($auditableRoles, $discordRole);
+            }
+        }
+
+        Log::info('DiscordAuditJob@reverse: Identified ' . sizeof($auditableRoles) . ' roles for audit.');
+
+        foreach ($auditableRoles as $auditableRole) {
+            $hmsRole = $roleRepository->findOneByDisplayName($auditableRole['name']);
+            $hmsUsers = $hmsRole->getUsers();
+
+            foreach ($hmsUsers as $hmsUser) {
+                if (! $hmsUser->getProfile()->getDiscordUserSnowflake()) {
+                    continue;
+                }
+
+                $discordMember = $discord->findMemberByProfile($hmsUser->getProfile());
+
+                if (! $discordMember) {
+                    if ($hmsRole->getName() == Role::MEMBER_CURRENT) {
```

Comment:
are u sure we should be clearing this?

Reply:
I think so :thinking:  If it gets to here then they are in HMS as a Current Member but there was no discord user found for them, either because they've left, they never joined the server, or the discord account never existed